java/src/main/java/com/logicalclocks/hsfs/FeatureView.java

```diff
@@ -0,0 +1,350 @@
+  }
+
+  /**
+   * Get all tags of the feature view.
+   *
+   * @return a map of tag name and values. The value of a tag can be any valid json - primitives, arrays or json objects
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  @JsonIgnore
+  public Map<String, Object> getTags() throws FeatureStoreException, IOException {
+    return featureViewEngine.getTags(this);
+  }
+
+  /**
+   * Get a single tag value of the feature view.
+   *
+   * @param name
+   *     name of the tag
+   * @return The value of a tag can be any valid json - primitives, arrays or json objects
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  @JsonIgnore
+  public Object getTag(String name) throws FeatureStoreException, IOException {
+    return featureViewEngine.getTag(this, name);
+  }
+
+  /**
+   * Delete a tag of the feature view.
+   *
+   * @param name
+   *     name of the tag to be deleted
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  public void deleteTag(String name) throws FeatureStoreException, IOException {
+    featureViewEngine.deleteTag(this, name);
+  }
+
+  public TrainingDatasetRepository getTrainingDataset(
+      Integer version, String description, Map<String, Float> splits,
+      StatisticsConfig statisticsConfig, Map<String, String> writeOptions
+  ) {
+    TrainingDataset trainingDataset = new FeatureStore().createTrainingDataset().build();
+    return featureViewEngine.getTrainingDataset(this, trainingDataset, writeOptions);
+  }
+
+  public TrainingDatasetRepository createTrainingDataset(
+      Integer version, String description, DataFormat dataFormat,
+      Boolean coalesce, StorageConnector storageConnector, String location, Map<String, Float> splits,
+      Long seed, StatisticsConfig statisticsConfig, Map<String, String> writeOptions
+  ) {
+    TrainingDataset trainingDataset = new FeatureStore().createTrainingDataset().build();
+    return featureViewEngine.createTrainingDataset(this, trainingDataset, writeOptions);
+  }
+
+  public void recreateTrainingDataset(Integer version) {
+    // TODO:
+  }
+
+  public void purgeTrainingData(Integer version) {
+    featureViewEngine.deleteTrainingDatasetOnly(this, version);
+  }
+
+  public void purgeAllTrainingData() {
+    featureViewEngine.deleteTrainingDatasetOnly(this);
+  }
+
+  public void deleteTrainingDataset(Integer version) {
+    featureViewEngine.deleteTrainingData(this, version);
+  }
+
+  public void deleteAllTrainingDatasets() {
+    featureViewEngine.deleteTrainingData(this);
+  }
+
+  /**
+   * Add name/value tag to the training dataset.
+   *
+   * @param name
+   *     name of the tag
+   * @param value
+   *     value of the tag. The value of a tag can be any valid json - primitives, arrays or json objects
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  public void addTrainingDatasetTag(String name, Object value, Integer version) throws FeatureStoreException,
```

Comment:
It's a nuance, but I would make version the first argument in the list. It's the order of the Python API and makes most sense semantically.